BezierKit/Library/PathComponent+Arc.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  PathComponent+Arc.swift
+//  BezierKit
+//
+//  Created by ky1vstar on 15.10.2020.
+//  Copyright © 2020 Holmes Futrell. All rights reserved.
+//
+
+import Foundation
+
+extension PathComponent {
+    public convenience init(
+        ovalIn rect: CGRect,
+        numberOfCurves: Int = 4
+    ) {
+        let diameter = min(rect.width, rect.height)
+        let radius = diameter / 2
+        
+        let tempPath = PathComponent(
+            arcCenter: .init(x: radius, y: radius),
+            radius: radius,
+            startAngle: 0,
+            endAngle: 2 * .pi,
+            clockwise: true
+            numberOfCurves: numberOfCurves
+        )
+        
+        let transform = CGAffineTransform.identity
+            .translatedBy(x: rect.origin.x, y: rect.origin.y)
+            .scaledBy(
+                x: rect.width / diameter,
+                y: rect.height / diameter
+            )
+        
+        self.init(
+            points: tempPath.points.map {
+                $0.applying(transform)
+            },
+            orders: tempPath.orders
+        )
+    }
+    
+    public convenience init(
+        arcCenter center: CGPoint,
+        radius: CGFloat,
+        startAngle: CGFloat,
+        endAngle: CGFloat,
+        clockwise: Bool,
+        numberOfCurves: Int = 4
```

Comment:
Let's make this `private` for now. In the future I'm going to add an API for the creation of `Path` objects that works similar to `CGMutablePath`. When that's ready we can use your code to add the functionality of `CGMutablePath.addArc(center: CGPoint, radius: CGFloat, startAngle: CGFloat, endAngle: CGFloat, clockwise: Bool, transform: CGAffineTransform = .identity)`